scripts/jobs/customer_services/vonage_landing_to_raw.py

```diff
@@ -1,213 +1,158 @@
+    largest_prefix = return_largest_prefix(subfolders)
 
-            importstring = path_dictionary["Prefix"]
+    subfolders = list_subfolders_in_directory(s3_client, bucket, largest_prefix)
+    largest_prefix = return_largest_prefix(subfolders)
 
-            importstring = re.search("[0-9]*\/$", importstring).group()
+    return largest_prefix
 
-            importstring = re.sub(string=importstring,
-                                  pattern="[^0-9.]".format(),
-                                  repl="")
-            list_of_raw_dates.append(importstring)
 
-        list_of_raw_dates = sorted(list_of_raw_dates, key=int, reverse=True)
-
-        largest_value = list_of_raw_dates[0]
-        return largest_value
-
-
-def get_latest_raw_zone_partition_date(s3_client, bucket, prefix):
-    # Get Year
-    folder_path = f'{prefix}/'
-
-    year_subfolders = list_subfolders_in_directory(s3_client, bucket, folder_path)
+def read_and_write_partition(landing_zone_bucket, partition_path, raw_zone_bucket, raw_zone_prefix):
+    key_import_date = find_importdate(partition_path)
+    import_day = key_import_date[6:]
+    import_month = key_import_date[4:6]
+    import_year = key_import_date[:4]
 
-    if year_subfolders == None:
-        print(f'No Files Found in Raw Zone. Will pull all Data')
-        latest_date = 0
-    else:
-        latest_year = get_latest_value(year_subfolders)
-        print(f'The Latest Year is {latest_year}')
+    print(f'Import Dates: {key_import_date}: Y{import_year} M{import_month} D{import_day}')
+    connection_path = f"s3://{landing_zone_bucket}/{partition_path}"
+    print(f'Connection Path: {connection_path}')
 
-        # Get Month
-        monthly_path = f'{folder_path}import_year={latest_year}/'
-        monthly_subfolders = list_subfolders_in_directory(s3_client, bucket, monthly_path)
-        latest_month = get_latest_value(monthly_subfolders)
+    dyf = glueContext.create_dynamic_frame.from_options(
+        connection_type="s3",
+        connection_options={"paths": [connection_path],
+                            "recurse": True
+                            },
+        format="json",
+        format_options={
+            "jsonPath": "$.items[*]",
+        }
+    )
 
-        daily_path = f'{monthly_path}import_month={latest_month}/'
-        daily_subfolders = list_subfolders_in_directory(s3_client, bucket, daily_path)
-        latest_day = get_latest_value(daily_subfolders)
+    print('Dynamic Frame Loaded')
 
-        latest_date = f'{latest_year}{latest_month}{latest_day}'
-        print(f'The Raw Import Date is {latest_date}')
+    print('Converting the DynamicFrame back to SparkDF')
+    dyf = dyf.toDF()
+    dyf = dyf.withColumn("import_date", lit(key_import_date))
+    dyf = dyf.withColumn("import_day", lit(import_day))
+    dyf = dyf.withColumn("import_month", lit(import_month))
+    dyf = dyf.withColumn("import_year", lit(import_year))
```

Comment:
you should chain these rather than evaluating after each one

dyf.withCoulumn(...).withCoulumn(...).withCoulumn(...)

I'd also consider assigning it to something that makes it clear you're no longer using a dynamic frame

Reply:
I think doing something like dyf = etc
then sparkDF = asdfljksdf(dyf) means I have both the dyf and the sparkDF in memory?